src/sensitivities/linalg/factorization/cholesky.jl

```diff
@@ -1,13 +1,11 @@
 import LinearAlgebra.BLAS: gemv, gemv!, gemm!, trsm!, axpy!, ger!
+import LinearAlgebra: cholesky
+import Base: getproperty
 
-# NOTE: Cholesky factorizations pose a significant issue for us as of Julia 0.7, since
-# the simple function chol, which produced the U in the factorization U'U, has been
-# deprecated in favor of accessing the .U field of a Cholesky object produced by cholesky.
-# This does not lend itself well to tracing. To get around this, we'll define our own
-# chol that users of Nabla can use to obtain the Julia 0.6 behavior.
-# See issue #105 for discussion.
-export chol
-chol(X::AbstractMatrix{<:Real}) = cholesky(X).U
+# Use the existing deprecation on Julia 0.7 and introduce a new one on later versions
```

Comment:
`Nabla.chol` was always a separate function here, so we should deprecate always.